developer_guides/topology2/topology2.rst

```diff
@@ -0,0 +1,895 @@
+  
+  .. code-block:: bash
+
+	# attribute qualifiers
+	attributes {
+		#
+		# This tells the compiler how to construct the object's name. For example, if the
+		# name attribute is set to "EQIIR-Coefficients", the object name will be
+		# constructed as "class_name.EQIIR-Coefficients"
+		#
+		!constructor [
+			"name"
+		]
+		#
+		# objects of the same class instantiated within the same alsaconf node must have unique
+		# name attribute
+		#
+		unique	"name"
+	}
+
+1.1.2 A Simple Class
+********************
+
+An example of a simple class definition with 2 attributes and qualifiers is as follows:
+
+.. code-block:: bash
+
+	Class.Base."data" {
+
+		# name for the data object
+		DefineAttribute."name" {
+			type	"string"
+		}
+
+		# bytes data
+		DefineAttribute."bytes" {
+			type	"string"
+		}
+
+		# attribute qualifiers
+		attributes {
+			#
+			# This tells the compiler how to construct the object's name. For example, if the
+			# name attribute is set to "EQIIR-Coefficients", the object name will be
+			# constructed as "data.EQIIR-Coefficients"
+			#
+			!constructor [
+				"name"
+			]
+			#
+			# data objects instantiated within the same alsaconf node must have unique
+			# name attribute
+			#
+			unique	"name"
+		}
+	}
+
+The "data" class definition contains 2 attributes namely, name and bytes, both of type "string".
+By default, all attributes of type "integer" unless specified otherwise like above. Currently,
+topology 2.0 supports only "integer" and "string" type attributes.
+
+The attribute qualifiers are used to describe how to instantiate an object from the class definition.
+
+In the above definition, The "constructor" array tells the compiler how to build the object's name.
+A data object instantiated with the name "EQIIR-Coefficients" will be given the name,
+"data.EQIIR-Coefficients" i.e. the class name followed by '.' followed by the attribute values
+separated by '.'.
+
+The "unique" qualifier signifies that 2 data objects instantiated within the same alsaconf node must
+have unique values for their name attribute. If two data objects are instantiated within the same alsaconf
+node with the same "name" attribute, there be no errors but the second object instance will override
+the first one.
+
+Let's consider another class definition example for the "pga" widget belonging to the class group "Widget".
+
+.. code-block:: bash
+
+	Class.Widget."pga" {
+		#
+		# Pipeline ID for the pga widget object
+		#
+		DefineAttribute."index" {}
+
+		#
+		# pga object instance
+		#
+		DefineAttribute."instance" {}
+		
+		# attribute qualifiers
+		attributes {
+			#
+			# The PGA widget name would be constructed using the index and instance
+			# attributes. For ex: "pga.1.1" or "pga.10.2" etc.
+			#
+			!constructor [
+				"index"
+				"instance"
+			]
+			
+			#
+			# pga widget objects instantiated within the same alsaconf node must have unique
+			# instance attribute
+			#
+			unique	"instance"
```

Comment:
that one is unclear: the instance can be understood as 'the instance within a pipeline' but the comment above hints at a different design 'the instance within a conf file'.

Which is it?